### PR TITLE
Power report 568

### LINKED ISSUE #568: PDS power data collection for ERC power report
There shall be a ROS service and/or node that when given the command, will capture data from the PDS and arm's smart servos to give us the ability to create a power report for each task of ERC. 

Please see the ERC 2022 requirements for the power report to note what we need to provide in the report.

### PR DESCRIPTION
**Pull request by vexandmore**
# Assignee Section

## Description
The power report node records power consumption starting when a start command is sent to it, and when stopped it saves it to a file. The latest power report is also retrievable from a service the node provides.

- Fixes on the pds node
- PDS simulator now sends more feeds
- Implemented PowerReportNode

### Steps for Testing
1. Ensure the rover is on this branch and started up (or use the local_rover.launch file in robot/util/rosRoverStart)
2. Run this command: ```rostopic pub /power_report_command std_msgs/String "data: 'start'"```. This will start the recording.
3. Wait some time
4. Run this command: ```rostopic pub /power_report_command std_msgs/String "data: 'stop'"```. This will stop the recording.
5. Check in ```~/Power_Reports``` for a power report.
6. Run this command: ```rosservice call /power_report_provider 0``` to see the latest report at any time

closes #568 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [x] Local Test Performed Successfully
- [x] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

**Review by RealWilliamWells — APPROVED** [review]
Amazing work. Your code is very clean and this was perfectly functional when we tested it yesterday evening

File: robot/rospackages/src/mcu_control/scripts/MockPdsNode.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def init_default_voluntary_param(parameters, default_value):
     # Check if parameters is a string (single parameter) or an array
     # (multiple parameters) and initialize appropriately
     if isinstance(parameters, str):
-        param_value = get_param_exist("PDS_mock_voltage")
+        param_value = get_param_exist("PDS_mock_voltage") # WTF? Why is PDS_mock_voltage hardcoded here? (Marc)
```
Comment (VimFreeman) [inline robot/rospackages/src/mcu_control/scripts/MockPdsNode.py:169]: Old code bad code

**Review by VimFreeman — APPROVED** [review]
Looks good at a glance. I would just check with physical measurement to make sure that all the math checks out before relying on it.

**Comment by RealWilliamWells** [comment]
> Looks good at a glance. I would just check with physical measurement to make sure that all the math checks out before relying on it.

We did do this using the OBC and it was spot on. Given that it's going through the same code for all of the other calculations, I believe it should be fine. Further testing will be done anyways this weekend as Joshua wants to measure current draw of the full system and we will be taking the rover outside a bit for filming

**Comment by VimFreeman** [comment]
> > Looks good at a glance. I would just check with physical measurement to make sure that all the math checks out before relying on it.
> 
> We did do this for the OBC and it was spot on. Given that it's going through the same code for all of the other calculations, I believe it should be fine. Further testing will be done anyways this weekend as Joshua wants to measure current draw of the full system and we will be taking the rover outside a bit for filming

Ok perfect

**Pull request merged**